catroid/src/org/catrobat/catroid/io/StorageHandler.java

```diff
@@ -685,7 +685,7 @@ private File copyFileBackPack(String programSubDirectory, String backpackSubDire
 			Log.e(TAG, "file " + inputFilePath + " doesn`t exist or can`t be read");
 			return null;
 		}
-		String inputFileChecksum = Utils.md5Checksum(inputFile).toUpperCase();
+		String inputFileChecksum = Utils.md5Checksum(inputFile).toUpperCase(Locale.getDefault());
```

Comment:
`If you want the methods to just perform ASCII replacement, for example to convert an enum name, call String#toUpperCase(Locale.US) instead.`

However, I think upper casing the checksum isn't necessary (or shouldn't be done) in the first place. ping @sjaindl